pallets/attesters/src/lib.rs

```diff
@@ -246,29 +320,102 @@ pub mod pallet {
                 None => T::DefaultCommission::get(),
             };
 
+            let next_index = Attesters::<T>::iter().count() as u32;
+
             Attesters::<T>::insert(
                 &account_id,
                 AttesterInfo {
                     key_ec: ecdsa_key,
                     key_ed: ed25519_key,
                     key_sr: sr25519_key,
                     commission,
+                    index: next_index,
                 },
             );
 
             // Self nominate in order to be part of the active set selection
-            Self::do_nominate(account_id.clone(), account_id, self_nominate_amount)?;
+            Self::do_nominate(account_id.clone(), account_id.clone(), self_nominate_amount)?;
+
+            Self::deposit_event(Event::AttesterRegistered(account_id.clone()));
+
+            Self::request_add_attesters_attestation(&account_id)?;
```

Comment:
We shouldnt deposit terminal events when things are fallible after, swap these and then remove the clone on the id

Reply:
addressed in [b45c033](https://github.com/t3rn/t3rn/pull/980/commits/b45c03309906d88f23d4c7710e6cba4c1498efd3)